Core/Src/cerb_utils.c

```diff
@@ -0,0 +1,30 @@
+/**
+ * @file cerb_utils.c
+ * @author Scott Abramson
+ * @brief Utility functions for Cerberus.
+ * @version 0.1
+ * @date 2024-08-04
+ * 
+ * @copyright Copyright (c) 2024
+ * 
+ */
+
+#include "cerb_utils.h"
+
+void debounce(bool input, osTimerAttr_t *timer, uint32_t period)
+{
+	if (input && !osTimerIsRunning(timer)) {
+		/* Start timer if a button has been pressed */
+		osTimerStart(timer, period);
+	} else if (!input && osTimerIsRunning(timer)) {
+		/* Button stopped being pressed. Kill timer. */
+		osTimerStop(timer);
+	}
+}
+
+osStatus_t queue_and_set_flag(osMessageQueueId_t queue, const void *msg_ptr,
```

Comment:
ngl I so completely disagree with the use of this.  Every individual should know freertos functions or in my opinion they should not have the ability or position on our team to be coding inside such threads.  We should not be abstracting two lines of code, especially when you need to use the actual two lines in the receiving part of the code.  Ok rant over feel free to ignore its not my gig.

Reply:
disagree with jack. doesnt have to do with hiding code from devs its simply a matter of abstracying functionality that is likely to be used in multilple places